File: gluecodium/src/test/resources/smoke/instances/output/android/com/example/smoke/SimpleInterfaceImpl.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
class SimpleInterfaceImpl extends NativeBase implements SimpleInterface {
    protected SimpleInterfaceImpl(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @NonNull
    public native String getStringValue();
    @NonNull
    public native SimpleInterface useSimpleInterface(@NonNull final SimpleInterface input);
}

File: gluecodium/src/test/resources/smoke/instances/output/android-kotlin/com/example/smoke/SimpleInterfaceImpl.kt
/*

 *
 */

@file:JvmName("SimpleInterfaceImpl")

package com.example.smoke

import com.example.NativeBase

/**
 * @suppress
 */
class SimpleInterfaceImpl : NativeBase, SimpleInterface {
    protected constructor(nativeHandle: Long, tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}

    override external fun getStringValue() : String
    override external fun useSimpleInterface(input: SimpleInterface) : SimpleInterface



    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}
